lib/charms/mongodb_libs/v0/mongodb_tls.py

```diff
@@ -129,7 +132,16 @@ def _on_tls_relation_broken(self, event: RelationBrokenEvent) -> None:
             return
 
         logger.debug("Restarting mongod with TLS disabled.")
-        self.charm.on_mongod_pebble_ready(event)
+        if self.substrate == "vm":
+            self.charm.unit.status = MaintenanceStatus("disabling TLS")
+            restart_mongod_service(
+                auth=True,
```

Comment:
Just a question: is there a situation where auth should be disabled and this value needs to be `False`?

Reply:
Great idea! I guess if someone is using the legacy relation and wants to use TLS, I've changed it now to use whatever the current `auth` is set to.